src/auth/connection.ts

```diff
@@ -289,17 +289,17 @@ export async function* loadLinkedProfilesIntoStore(
             name,
             type: 'iam',
             subtype: 'linked',
-            ssoSession: source,
+            ssoSession: sourceId,
             ssoRoleName: info.roleName,
             ssoAccountId: info.accountId,
         })
 
         yield [id, profile] as const
     }
 
-    const isBuilderId = startUrl === builderIdStartUrl // Special case.
-    if (!isBuilderId && (accounts.size === 0 || found.size === 0)) {
-        const name = truncateStartUrl(startUrl)
+    const hasScopes = !!ssoProfile.scopes?.some(s => !ssoAccountAccessScopes.includes(s))
```

Comment:
Should the `!` not be in `!ssoAccountScopes...` since this implies the ssoProfile has no `ssoAccountAccessScopes` and `hasScopes` means the ssoProfile has `ssoAccountAccessScopes`?

Reply:
> since this implies the ssoProfile has no `ssoAccountAccessScopes`

The query is asking "are there any scopes which are _not_ `ssoAccountAccessScopes`"? But it's hard to read, will try to improve it.

Reply:
Added a comment, but I can't find a better way to express the query. Also considered renaming `hasScopes` but nothing else seems better.